File: backend/app/models/invoice.py
import secrets
from datetime import datetime


def new_invoice(
    client_id: str,
    amount_cents: int,
    line_items: list[dict],
    due_date: datetime,
    project_id: str | None = None,
    stripe_invoice_id: str = "",
) -> dict:
    return {
        "client_id": client_id,
        "project_id": project_id,
        "stripe_invoice_id": stripe_invoice_id,
        "amount_cents": amount_cents,
        "status": "draft",
        "due_date": due_date,
        "line_items": line_items,
        "token": secrets.token_urlsafe(24),
        "created_at": datetime.utcnow(),
        "paid_at": None,
    }

File: backend/src/models/invoice.ts
import crypto from 'crypto';

export function newInvoice(
  clientId: string,
  amountCents: number,
  lineItems: { description: string; amount_cents: number; quantity: number }[],
  dueDate: Date,
  projectId: string | null = null,
  stripeInvoiceId = '',
) {
  return {
    client_id: clientId,
    project_id: projectId,
    stripe_invoice_id: stripeInvoiceId,
    amount_cents: amountCents,
    status: 'draft',
    due_date: dueDate,
    line_items: lineItems,
    token: crypto.randomBytes(24).toString('base64url'),
    created_at: new Date(),
    paid_at: null as Date | null,
  };
}
